.eslintrc.js

```diff
@@ -1,9 +1,15 @@
 const path = require('path');
 
-const forbitTopLevelMessage = [
+const forbidTopLevelMessage = [
   'Prefer one level nested imports to avoid bundling everything in dev mode',
   'See https://github.com/mui-org/material-ui/pull/24147 for the kind of win it can unlock.',
 ].join('\n');
+// This only applies to packages published from this monorepo.
+// If you build a library around `@material-ui/core` you can safely use `createStyles` without running into the same issue as we are.
+const forbidCreateStylesMessage =
+  'Use `Styles<Theme, Props, ClassKey>` instead if the styles are exported. Otherwise use `as const` assertions. ' +
```

Comment:
```suggestion
  'Use `MuiStyles<ClassKey, Props>` instead if the styles are exported. Otherwise use `as const` assertions. ' +
```

Recommend convenience type.